cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -1451,6 +1451,60 @@ public void testCruiseControlDisabledToEnabledBehaviour(VertxTestContext context
                 }));
     }
 
+    @Test
+    public void testKrWhenCCUserTaskIsEmpty(VertxTestContext context) throws IOException, URISyntaxException {
+
+        cruiseControlServer.setupCCRebalanceResponse(0, CruiseControlEndpoints.REBALANCE);
+
+        KafkaRebalance kr = new KafkaRebalanceBuilder(createKafkaRebalance(namespace, CLUSTER_NAME, RESOURCE_NAME, EMPTY_KAFKA_REBALANCE_SPEC, false))
+                .withNewStatus()
+                .withObservedGeneration(1L)
+                .withConditions(new ConditionBuilder()
+                        .withType("ProposalReady")
+                        .withStatus("True")
+                        .build())
+                .endStatus()
+                .build();
+
+        Crds.kafkaRebalanceOperation(client).inNamespace(namespace).resource(kr).create();
+        Crds.kafkaRebalanceOperation(client).inNamespace(namespace).resource(kr).updateStatus();
+
+        crdCreateKafka();
+        crdCreateCruiseControlSecrets();
+
+        Checkpoint checkpoint = context.checkpoint();
+        krao.reconcile(new Reconciliation("test-trigger", KafkaRebalance.RESOURCE_KIND, namespace, RESOURCE_NAME))
+                .onComplete(context.succeeding(v -> context.verify(() -> {
+                    // the resource moved to ProposalReady
+                    assertState(context, client, namespace, RESOURCE_NAME, KafkaRebalanceState.ProposalReady);
+                })))
+                .compose(v -> {
+                    if (!Annotations.booleanAnnotation(kr, Annotations.ANNO_STRIMZI_IO_REBALANCE_AUTOAPPROVAL, false)) {
+                        annotate(client, namespace, kr.getMetadata().getName(), KafkaRebalanceAnnotation.approve);
+                    }
```

Comment:
When would this be true / false? I do not see any parametrization (but maybe I missed it?).

Reply:
Yeah, you don't need the if, just annotate. Alternatively, you can edit the KR at creation time and add the auto-approval annotation there, which I think it's better.